### DISCUSSION TITLE
is there a support to specify variants in the config of tailwind css v3

### DISCUSSION THREAD

**Discussion by neel1998** [tailwindlabs/tailwindcss#12829] [Help] [answered]
in tailwindcss v1, we could define the specific selectors and variants we would want for each of the classes,

```
variants: {
    borderRadius: ['hover'],
  },
```

is there someway we can do this in tailwind v3. My use case is that i want to generate a default css file based on the config. so I am safelisting all the classes,

```
safelist: [
    {
      pattern: /./,
    },
  ],
  ```
  and then I am overriding theme and turning off core plug ins to just generate my desire classes. However I am not able to get hover, active and screen variants classes generated.
  
 Is there a solution for this?

**Comment by wongjn** [comment, marked as answer]
You can add variants to regular expression safelists [via the `variants` key](https://tailwindcss.com/docs/content-configuration#using-regular-expressions):
```js
safelist: [
  {
    pattern: /./,
    variants: [
      'hover',
      'active',
      'sm',
      …
    ],
  },
],
```

**Reply by neel1998** [reply]
thanks a lot! this works
